packages/expo-camera/ios/EXCamera/EXCamera.m

```diff
@@ -534,20 +534,24 @@ - (void)record:(NSDictionary *)options resolve:(UMPromiseResolveBlock)resolve re
     UM_WEAKIFY(self);
     dispatch_async(self.sessionQueue, ^{
       UM_STRONGIFY(self);
-      if (!self) {
-        reject(@"E_IMAGE_SAVE_FAILED", @"Camera view has been unmounted.", nil);
-        return;
-      }
-      if (!self.fileSystem) {
-        reject(@"E_IMAGE_SAVE_FAILED", @"No file system module", nil);
-        return;
+      // it is possible that the session has been invalidated at this point
+      // for example, the video codec option is invalid and so this call has already rejected
+      if (self.movieFileOutput != nil) {
```

Comment:
is the promise always guaranteed to be rejected at this point if movieFileOutput is nil? this seems potentially fragile if not, we would leave a promise dangling

Reply:
that's a very good point - thank you! I'll rework this